cadc-access-control-server/src/main/java/ca/nrc/cadc/auth/AuthenticatorImpl.java

```diff
@@ -147,6 +149,21 @@ public void augmentSubject(final Subject subject)
             PluginFactory pluginFactory = new PluginFactory();
             UserPersistence userPersistence = pluginFactory.createUserPersistence();
             Principal ldapPrincipal = getLdapPrincipal(subject);
+
+            // Remove the HttpPrincipal in subject, if any
+            Principal toBeRemoved = null;
+            Set<Principal> principals = subject.getPrincipals();
+            for (Iterator<Principal> iterator = principals.iterator(); iterator.hasNext();) {
+                Principal princ = iterator.next();
+                if (princ instanceof HttpPrincipal) {
+                    toBeRemoved = princ;
+                    break;
+                }
+            }
+            if (toBeRemoved != null) {
+                subject.getPrincipals().remove(toBeRemoved);
+            }
+
```

Comment:
Like we discussed, I think all you need to do is, at this same place in the code:
`subject.getPrincipals().clear()`

Reply:
Actually, only removing the HttpPrincipals would be safer.  So:
`subject.getPrincipals(HttpPrincipal.class).clear()`

Reply:
Right. Thanks. Done.